File: 49. Group Anagrams/49. Group Anagrams.dart
class Solution {
  List<List<String>> groupAnagrams(List<String> strs) {
    Map<String, List<String>> anagramGroups = {};
    
    for (String str in strs) {
      List<String> charList = str.split('')..sort();
      String sortedStr = charList.join();
      
      if (!anagramGroups.containsKey(sortedStr)) {
        anagramGroups[sortedStr] = [];
      }
      anagramGroups[sortedStr]!.add(str);
    }
    
    return anagramGroups.values.toList();
  }
}

File: 49. Group Anagrams/49. Group Anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var anagramGroups = new Dictionary<string, List<string>>();
        
        foreach (var str in strs) {
            var sortedStr = new string(str.OrderBy(c => c).ToArray());
            if (!anagramGroups.ContainsKey(sortedStr)) {
                anagramGroups[sortedStr] = new List<string>();
            }
            anagramGroups[sortedStr].Add(str);
        }
        
        return new List<IList<string>>(anagramGroups.Values);
    }
}